### PR TITLE
chore(travis): move deploy command to a separate script

### PR DESCRIPTION
**Pull request by jviotti**
We're passing a shell conditional to `deploy.script`, however Travis CI
seems to get confused about this in the deploy section, causing
GNU/Linux to run `make publish-aws-s3` directly on the Travis CI build
instead of in the Docker container, which causes the deployment to
eventually fail because of missing dependencies.

According to Travis CI:

> `deploy.script` must be a scalar pointing to an executable file or
> command.

See https://travis-ci.org/resin-io/etcher/jobs/214723725 for a failure
example.

We also move the Appveyor deploy script to a separate file for symmetry
purposes.

Signed-off-by: Juan Cruz Viotti <jviotti@openmailbox.org>

### CODE REVIEW

File: scripts/ci/deploy-travis.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/bin/sh
```
Comment (lurch) [inline scripts/ci/deploy-travis.sh]: This should probably be `#!/bin/bash` for consistency with all our other scripts

File: scripts/ci/deploy-travis.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###
+
+set -e
+set -u
+
```
Comment (lurch) [inline scripts/ci/deploy-travis.sh]: Would it be worth including a dependency-check for `make` ?

File: scripts/ci/deploy-travis.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
+  ./scripts/build/docker/run-command.sh -r $TARGET_ARCH -s $PWD -c "make publish-aws-s3"
```
Comment (lurch) [inline scripts/ci/deploy-travis.sh]: This should probably use the same "$HERE/...etc..." that other scripts use?

(Hmmm, would it be worth submitting a separate PR to update _all_ our scripts to use the "$HERE/...etc..." style of referencing paths?)

**Comment by lurch** [comment]
Ahhh, if Travis is indeed running `make publish-aws-s3` instead of `./scripts/build/docker/run-command.sh -r ${TARGET_ARCH} -s ${PWD} -c "make publish-aws-s3"`, that would explain the weird `Dependency missing: bower` error. Good job @jviotti !

Doesn't really make much difference, but for the sake of symmetry would it be worth adding a `scripts/ci/deploy-appveyor.sh` too?

File: scripts/ci/deploy-travis.sh (hunk 1 above)
Comment (jviotti) [inline scripts/ci/deploy-travis.sh]: Done!

File: scripts/ci/deploy-travis.sh (hunk 2 above)
Comment (jviotti) [inline scripts/ci/deploy-travis.sh]: Makes sense, done!

File: scripts/ci/deploy-travis.sh (hunk 3 above)
Comment (jviotti) [inline scripts/ci/deploy-travis.sh]: > (Hmmm, would it be worth submitting a separate PR to update all our scripts to use the "$HERE/...etc..." style of referencing paths?)

Yeah, lets do that instead!

**Comment by jviotti** [comment]
:) Done!

File: scripts/ci/deploy-travis.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+# limitations under the License.
+###
+
+set -e
+set -u
+
+./scripts/build/check-dependency.sh make
+
+if [ "$TRAVIS_OS_NAME" == "linux" ]; then
+  ./scripts/build/docker/run-command.sh -r $TARGET_ARCH -s $PWD -c "make publish-aws-s3"
```
Comment (lurch) [inline scripts/ci/deploy-travis.sh]: We should quote these variables, and `$PWD` might be better as `"$(pwd)"`

(and the variables in `.travis.yml` ought to be quoted too)

**Comment by lurch** [comment]
Should we make these `deploy...` scripts fail if `$CI` isn't set to 1?
And/or we could make the deploy-travis script fail if it doesn't detect the Travis-specific envvars, and make the deploy-appveyor script fail if it doesn't detect the Appveyor-specific envvars?

File: scripts/ci/deploy-travis.sh (hunk 4 above)
Comment (jviotti) [inline scripts/ci/deploy-travis.sh]: Fixed!

**Comment by jviotti** [comment]
> Should we make these deploy... scripts fail if $CI isn't set to 1?
And/or we could make the deploy-travis script fail if it doesn't detect the Travis-specific envvars, 

It will fail out of the box given that we use `set -u`. Do you think we should still add specific conditionals in order to fail with a better error message?

> and make the deploy-appveyor script fail if it doesn't detect the Appveyor-specific envvars?

I'll add a conditional there.

**Comment by lurch** [comment]
IMHO it'd be better to explicitly check (as you're now doing in the appveyor script), rather than "hoping" that `set -u` always remains set and using that to catch any errors.

**Comment by jviotti** [comment]
Sounds good. Fixed!
